.github/workflows/test-image-execution.yaml

```diff
@@ -0,0 +1,124 @@
+name: Test Image execution on OS
+
+on:
+  # run it on push to the default repository branch
+  push:
+    branches: [master]
+  pull_request:
+  # Only trigger, when the build workflow succeeded
+  # workflow_run:
+  #   workflows: [ "Deploy Application images when Merging on Master" ]
+  #   types: [completed]
+
+env:
+  LISTINGAPI_ENDPOINT: "http://localhost:8181/listings/32682"
+  PRICEMAP_PYTHON_ENDPOINT: "http://localhost:8080"
+  PRICEMAP_TYPESCRIPT_ENDPOINT: "http://localhost:8080"
+
+jobs:
+  build:
+    name: Test containers execution
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+    services:
+      db:
+        image: "mdillon/postgis:11-alpine"
+        env:
+          POSTGRES_DB: pricemap
+          POSTGRES_USER: pricemap
+          POSTGRES_PASSWORD: pricemap
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 20s
+          --health-timeout 5s
+          --health-retries 5
+          --health-start-period 15s
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
+
+    # steps to perform in job
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Initialize Database
+        run: "psql -f pricemap-python/docker/db/init.sql postgresql://pricemap:pricemap@localhost:5432/pricemap"
+
+      - name: Create Network
+        run: |
+          docker network create github_network
+
+      - name: Create listingAPI container
+        id: create_listingapi_container
+        run: |
+          docker run --name listingapi --network github_network --network-alias listingapi -p 8181:5000 --detach \
+          -e "FLASK_APP=app" -e GITHUB_ACTIONS=true -e \
+          CI=true ghcr.io/meilleursagents/backend-technical-test/listingapi:latest flask run -h 0.0.0.0
+
+      - name: Create pricemap Python container
+        id: create_pricemap_python_container
+        run: |
+          docker run --name pricemap-python --network github_network --network-alias pricemap-python -p 8080:5000 \
+          --detach -e "PGHOST=db" -e "PGDATABASE=pricemap" -e "PGUSER=pricemap" -e "PGPASSWORD=pricemap" \
```

Comment:
`db` don't resolve in this network. You need to use `host.docker.internal` to get the host ip, where the port 5432 is binded to the db container port.